src/libs/nav2_tasks/include/nav2_tasks/task_client.hpp

```diff
@@ -70,6 +70,24 @@ class TaskClient
     cancelPub_->publish(msg);
   }
 
+  bool waitForServer(std::chrono::milliseconds timeout = std::chrono::milliseconds(3000))
+  {
+    std::string taskName = getTaskName<CommandMsg, ResultMsg>();
+    taskName += "_command";
+
+    auto startTime = std::chrono::high_resolution_clock::now();
+
+    while (node_->count_subscribers(taskName) < 1) {
```

Comment:
I don't think this is the right way to way for a service in the long run. Maybe at least include a TODO here to remind us to replace it with something better eventually.